examples/zero_shot_instance_segmentation.ipynb

```diff
@@ -0,0 +1,441 @@
+    "\n",
+    "We will also use the headless version of [OpenCV](https://opencv.org/) to convert the videos to frame images.\n",
+    "\n",
+    "**Note**: For image-only datasets, FiftyOne [natively supports SAM](https://docs.voxel51.com/user_guide/model_zoo/models.html#segment-anything-vith-torch) as part of the [FiftyOne Model Zoo](https://docs.voxel51.com/user_guide/model_zoo/index.html#)!"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's install the neccessary libraries:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install fiftyone transformers ultralytics opencv-python-headless"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And then import all of the necessary packages:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from glob import glob\n",
+    "import os\n",
+    "\n",
+    "import cv2\n",
+    "import fiftyone as fo\n",
+    "import numpy as np\n",
+    "from PIL import Image\n",
+    "from transformers import pipeline\n",
+    "from ultralytics import SAM"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Download the dataset from [this zip file](https://drive.google.com/file/d/18okj067dIOXgeHu47bsKAtaq56jRWisQ/view) to a folder `taranaki` and unzip it."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then load it into FiftyOne:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = fo.Dataset.from_dir(\n",
+    "    \"taranaki\", \n",
+    "    dataset_type=fo.types.FiftyOneVideoLabelsDataset\n",
+    "    )"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then we can give it a name and make it persistent:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.name = \"wildlife-watcher\"\n",
+    "dataset.persistent = True"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then we convert the videos into sequences of images and save them in a folder. We will also use FiftyOne's `ensure_frames()` method to ensure that the frames are accessible on the videos, so we can add predictions to them."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.ensure_frames()\n",
```

Comment:
It looks like this ensure_frames() call isn't necessary until later; conceptually seems a bit out of place here as you are only sampling the vids here